### PR TITLE
added collections endpoint

### PR DESCRIPTION
**Pull request by pgulley**
Simple addition of a /collections endpoint, which lists the available collections on the service.

### CODE REVIEW

**Review by thepsalmist — APPROVED** [review]
LGTM

**Review by kilemensi — APPROVED** [review]

**Comment by pgulley** [comment]
Given conversation elsewhere, realizing that that it would be preferable to actually query the index rather than return the hard-coded list... Will circle back.

File: api.py [hunk 1]
```diff
@@ -287,6 +287,13 @@ def version_root(req: Request):
     return "\n".join(['<ul>'] + lis + ['</ul>'])
 
 
+@v1.get("/collections", tags=["data"])
+@v1.head("/collections", include_in_schema=False)
+def get_collections(req:Request):
+    return [i for index in ES.indices.get('*')]
```
Comment (thepsalmist) [inline api.py:293]: For search purposes don't you think we should be getting indices prefixed with our ELASTICSEARCH_INDEX_PREFIX, a `*` may result in `dirty indices` or indices created by certain API actions

**Review by philbudne — APPROVED** [review]

**Comment by pgulley** [comment]
Now ready for actual final review- apologies for the premature request

**Review by thepsalmist — APPROVED** [review]

File: api.py [hunk 2]
```diff
@@ -34,9 +34,19 @@ class ApiVersion(str, Enum):
...
 
+ELASTICSEARCH_INDEX_NAME_PREFIX = os.getenv("ELASTICSEARCH_INDEX_NAME_PREFIX")
+
 ES = Elasticsearch(config["eshosts"], **config["esopts"])
 
-Collection = list_to_enum("Collection", config["indexes"])
+
+def get_allowed_collections():
+    #Only expose indexes with the correct prefix, and add a wildcard as well. 
+    all_indexes = [index for index in ES.indices.get(index='*') if ELASTICSEARCH_INDEX_NAME_PREFIX in index]
```
Comment (kilemensi) [inline api.py:44]: Since it's a prefix, may be
```python
index.startswith(ELASTICSEARCH_INDEX_NAME_PREFIX)
```
is better?

File: api.py [hunk 3]
```diff
@@ -34,9 +34,19 @@ class ApiVersion(str, Enum):
...
+    all_indexes.extend([f"{ELASTICSEARCH_INDEX_NAME_PREFIX}_*)"])
```
Comment (kilemensi) [inline api.py:45]:
```suggestion
    all_indexes.append(f"{ELASTICSEARCH_INDEX_NAME_PREFIX}_*)")
```
I take we're are only adding this one item.

**Review by kilemensi — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,78 +1,44 @@
...
 
-An API wrapper to the Elasticsearch index of web archival collections and a web UI to explore those indexes.
+An API wrapper to the Elasticsearch index of web archival collections and a web UI to explore those indexes. A part of the [story-indexer stack](https://github.com/mediacloud/story-indexer). Mantained as a separate repository for future legibility. 
 
 ## ES Index
 
-The API service expects the following ES index schema, where `title` and `snippet` fields must have the `fielddata` enabled (if they have the type `text`) and `first_captured` field stores 14-digit datetime in the format `YYYYMMDDhhmmss`.
-
+The API service expects the following ES index schema, where `title` and `snippet` fields must have the `fielddata` enabled (if they have the type `text`).
+This is currently defined in the story-indexer stack, but is replicated here for convenience.
```
Comment (philbudne) [inline README.md:8]: Replication seems.... less than ideal!  BUT given that they're in different repos, an easy/clean solution seems.... elusive!

Comment (pgulley) [reply README.md:8]: Agreed- the concerns are a little muddled. This seemed like the best way to document how what dependencies exist given how things are laid out now.

**Pull request merged**